src/Microsoft.PowerShell.Commands.Diagnostics/GetEventCommand.cs

```diff
@@ -1538,6 +1542,41 @@ private string HandleDataHashValue(Object value)
 
 
         //
+        // HandleNamedDataHashValue helper for hashtable structured query builder.
+        // Constructs and returns named event data field XPath portion as a string.
+        // Fix Issue #2327
+        //
+        private string HandleNamedDataHashValue(String key, Object value)
+        {
+            string ret = "";
+            if (value is Array)
+            {
+                Array dataArray = (Array)(value);
+                ret += "(";
```

Comment:
@daxian-dbw I oppose for two reasons:
1. As mentioned, the code is very old and needed to be revised entirely. I don't like the idea of disguising one hole out of thirty.
2. Soon I plan to expand the cmdlet and add support Suppress. To do this, I need to completely rewrite the code that builds a query. #2402  
So I don't want to do the work twice. To solve #2402  I will already use the new technique.